prettyPrinters/providers.py

```diff
@@ -351,25 +363,28 @@ class StdRcSyntheticProvider:
     """Pretty-printer for alloc::rc::Rc<T> and alloc::sync::Arc<T>
 
     struct Rc<T> { ptr: NonNull<RcBox<T>>, ... }
-    struct NonNull<T> { pointer: NonZero<*const T> }
+    rust 1.31.1: struct NonNull<T> { pointer: NonZero<*const T> }
+    rust 1.33.0: struct NonNull<T> { pointer: *const T }
     struct NonZero<T>(T)
     struct RcBox<T> { strong: Cell<usize>, weak: Cell<usize>, value: T }
     struct Cell<T> { value: UnsafeCell<T> }
     struct UnsafeCell<T> { value: T }
 
     struct Arc<T> { ptr: NonNull<ArcInner<T>>, ... }
-    struct NonNull<T> { pointer: NonZero<*const T> }
-    struct NonZero<T>(T)
     struct ArcInner<T> { strong: atomic::AtomicUsize, weak: atomic::AtomicUsize, data: T }
     struct AtomicUsize { v: UnsafeCell<usize> }
-    struct UnsafeCell<T> { value: T }
     """
 
     def __init__(self, valobj, dict, is_atomic=False):
         # type: (SBValue, dict, bool) -> StdRcSyntheticProvider
         self.valobj = valobj
 
-        self.ptr = self.valobj.GetChildMemberWithName("ptr").GetChildMemberWithName("pointer").GetChildAtIndex(0)
+        self.ptr = self.valobj.GetChildMemberWithName("ptr").GetChildMemberWithName("pointer")
+        # https://github.com/rust-lang/rust/commit/7a0911528058e87d22ea305695f4047572c5e067
+        # TODO: get rid of it after new rustc stable release
+        if not self.ptr.TypeIsPointerType():
+            self.ptr = self.ptr.GetChildAtIndex(0)
```

Comment:
Can we extract this code into a separate method?
At this moment we have 3 copy of it